source/daplink/interface/swd_host.c

```diff
@@ -878,6 +897,11 @@ uint8_t swd_set_target_state_hw(TARGET_RESET_STATE state)
             os_dly_wait(2);
             swd_set_target_reset(0);
             os_dly_wait(2);
+            
+            // Enable debug again because TARGET RESET might have disabled it
+            if (!swd_write_word(DBG_HCSR, DBGKEY | C_DEBUGEN)) {
+                return 0;
+            }
```

Comment:
This line is causing tests to fail. The sequence here is to turn on the reset vector catch and then perform a hardware reset so the device is halted right at boot. By writing to the DHCSR here the C_HALT bit gets written to 0 causing the processor to start running. Is this change required for the NRF51 or NRF52 to work correctly?